### ISSUE TITLE
Running on MS Windows, can't get final result

### ISSUE THREAD

**Issue by akulubala** [clue/graph-uml#30]
Hi,i'm trying to use this tool,after i install all needed,i try to run example/graph-self.php,it's takes a long time to exculte php script but nothing finally happend..then i debuged it,it stucked in GraphViz.php line 125:

```
    if (strtoupper(substr(PHP_OS, 0, 3)) === 'WIN') {
        echo "ausgabe\n";
        //exec($tmp . ' >NUL');
        exec($tmp);
   }
```

$tmp is something like :C:\WINDOWS\Temp\gra163.tmp.svg
gra163.tmp.svg file contents are : 

```
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN"
"http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">
<!-- Generated by graphviz version 2.36.0 (20140111.2315)
 -->
<!-- Title: G Pages: 1 -->
<svg width="458pt" height="274pt"
 viewBox="0.00 0.00 458.00 274.00" xmlns="http://www.w3.org/2000/svg"     xmlns:xlink="http://www.w3.org/1999/xlink">
<g id="graph0" class="graph" transform="scale(1 1) rotate(0) translate(4 270)">
<title>G</title>
<polygon fill="white" stroke="none" points="-4,4 -4,-270 454,-270 454,4 -4,4"/>
<!-- Fhaculty\\Graph\\Uml\\ClassDiagramBuilder -->
<g id="node1" class="node">    <title>Fhaculty\\Graph\\Uml\\ClassDiagramBuilder</title>
<polygon fill="none" stroke="black" points="0,-1 0,-265 450,-265 450,-1 0,-1"/>
<text text-anchor="middle" x="225" y="-249.8" font-family="Times New Roman,serif"     font-size="14.00">Fhaculty\Graph\Uml\ClassDiagramBuilder</text>
<polyline fill="none" stroke="black" points="0,-242 450,-242 "/>
<text text-anchor="middle" x="225" y="-226.8" font-family="Times New Roman,serif"     font-size="14.00"> </text>
<polyline fill="none" stroke="black" points="0,-219 450,-219 "/>
<text text-anchor="start" x="8" y="-203.8" font-family="Times New Roman,serif" font-    size="14.00">+ __construct(graph : Fhaculty\Graph\Graph)</text>
<text text-anchor="start" x="8" y="-188.8" font-family="Times New Roman,serif" font-    size="14.00">+ setOption(name, flag)</text>
<text text-anchor="start" x="8" y="-173.8" font-family="Times New Roman,serif" font-    size="14.00">+ hasClass(class)</text>
<text text-anchor="start" x="8" y="-158.8" font-family="Times New Roman,serif" font-        size="14.00">+ createVertexClass(class)</text>
<text text-anchor="start" x="8" y="-143.8" font-family="Times New Roman,serif" font-size="14.00">+ createVertexExtension(extension)</text>
<text text-anchor="start" x="8" y="-128.8" font-family="Times New Roman,serif" font-   size="14.00"># getLabelRecordClass(reflection : ReflectionClass) : string</text>
<text text-anchor="start" x="8" y="-113.8" font-family="Times New Roman,serif" font-    size="14.00"># getLabelRecordExtension(reflection : ReflectionExtension) :     string</text>
<text text-anchor="start" x="8" y="-98.8" font-family="Times New Roman,serif" font-    size="14.00"># getLabelRecordConstants(reflection : mixed) : string</text>
<text text-anchor="start" x="8" y="-83.8" font-family="Times New Roman,serif" font-    size="14.00"># getLabelRecordFunctions(functions : mixed, class : mixed = NULL) :     string</text>
<text text-anchor="start" x="8" y="-68.8" font-family="Times New Roman,serif" font-    size="14.00"># isVisible(reflection : mixed) : bool</text>
<text text-anchor="start" x="8" y="-53.8" font-family="Times New Roman,serif" font-    size="14.00">+ createVertexNote(note, for = NULL) : LoaderUmlClassDiagram</text>
<text text-anchor="start" x="8" y="-38.8" font-family="Times New Roman,serif" font-    size="14.00">+ createGraphComponent(class) : Graph</text>
<text text-anchor="start" x="8" y="-23.8" font-family="Times New Roman,serif" font-    size="14.00">+ createGraphsComponents() : mixed</text>
<text text-anchor="start" x="8" y="-8.8" font-family="Times New Roman,serif" font-    size="14.00">+ getNumberOfComponents() : int</text>
</g>
</g>
</svg>
```

can you please tell me how to run it correctly?
i installed GraphViz 2.36,and set it path to environment path..
Thank you..

**Comment by clue** [comment]
Thanks for reporting, I'll look into this.

Admittedly, MS Windows has never been a main target of this library, so I'm looking forward to assist in making it run smoothly.

**Comment by clue** [comment]
Alright, I've just released a tagged clue/graph v0.7.1 with the relevant changes (clue/graph#99). Also, the reference in the lock file has been updated, so you'll have to update your dependencies (`composer install`).

I've tested this on Windows XP with Graphviz 2.34, but didn't bother modifying my PATH. Instead, I've patched the `example/graph-self.php` like this:

``` php
$graphviz->setExecutable('C:\Programme\Graphviz2.34\bin\dot');
$graphviz->setFormat('png');
```

Feel free to report back.
